### ISSUE TITLE
CA2000: Rule does not trigger for IDisposable objects returned from methods with a Tuple

### ISSUE THREAD

**Issue by nick-morhun** [dotnet/roslyn-analyzers#7714]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 9.0.0

### Describe the bug

False negative for the following code:

```
using System;
using System.IO;

public static class Class1
{
  public static void Example(int x)
  {
    var (toDispose, _) = GetByZ(x); // No detection
    Console.Write(toDispose.ToString());
  }

  private static (MemoryStream, int) GetByZ(int z)
  {
    return (new MemoryStream(), z + 1);
  }
}
```

### Expected behavior

Detection for the toDispose variable inside the Example() method.

### Actual behavior

No detection.
